pandas-stubs/core/tools/numeric.pyi

```diff
@@ -1 +1,64 @@
-def to_numeric(arg, errors: str = ..., downcast=...): ...
+from typing import (
+    Literal,
+    overload,
+)
+
+import numpy as np
+import pandas as pd
+
+from pandas._typing import (
+    IgnoreRaiseCoerce,
+    Scalar,
+    npt,
+)
+
+@overload
+def to_numeric(
+    arg: Scalar,
+    errors: Literal["raise", "coerce"] = ...,
+    downcast: Literal["integer", "signed", "unsigned", "float"] | None = ...,
+) -> int | float: ...
+@overload
+def to_numeric(
+    arg: Scalar,
+    errors: Literal["ignore"],
+    downcast: Literal["integer", "signed", "unsigned", "float"] | None = ...,
+) -> Scalar: ...
+@overload
+def to_numeric(
+    arg: list | tuple | npt.NDArray,
+    errors: IgnoreRaiseCoerce = ...,
+    *,
+    downcast: Literal["integer", "signed", "unsigned", "float"],
+) -> npt.NDArray: ...
+@overload
+def to_numeric(
+    arg: list | tuple | npt.NDArray,
+    errors: Literal["ignore"],
+    downcast: None = ...,
+) -> npt.NDArray: ...
+@overload
+def to_numeric(
+    arg: list | tuple | npt.NDArray,
+    errors: Literal["raise", "coerce"] = ...,
+    downcast: None = ...,
+) -> npt.NDArray[np.intp] | npt.NDArray[np.float_]: ...
+@overload
+def to_numeric(
+    arg: pd.Series,
+    errors: IgnoreRaiseCoerce = ...,
+    *,
+    downcast: Literal["integer", "signed", "unsigned", "float"],
+) -> pd.Series: ...
+@overload
+def to_numeric(
+    arg: pd.Series,
+    errors: Literal["ignore"],
+    downcast: None = ...,
+) -> pd.Series: ...
+@overload
+def to_numeric(
+    arg: pd.Series,
+    errors: Literal["raise", "coerce"] = ...,
+    downcast: None = ...,
+) -> pd.Series[int] | pd.Series[float]: ...
```

Comment:
We might be better off just returning `pd.Series` here, because when we return `Union` types, we then might require people to do `cast` operations.

Reply:
Feels like the enhanced specificity helps rather than Series[Any]

Reply:
I see your point.  Right now, we don't have any methods in pandas that differentiate between requiring `Series[int]` and `Series[float]` as a parameter, so I guess this is OK for now.

Reply:
On the other hand, as a user, you could have:
```python
    ss = pd.to_numeric(pd.Series([1, 2, 3]))
    reveal_type(ss)
    q = ss[0]
    reveal_type(q)
    reveal_type(ss.loc[1])
```
here `q` and `ss.loc[1]` will have type `int | float` (with pyright), which could be inconvenient and require a `cast`.  But I think you'd have to do a `cast` if we declared it as `Series[float]` or just `Series`.

I'm willing to leave it as you have it.